toplevel/toploop.ml

```diff
@@ -95,7 +104,11 @@ let load_file = load_file false
 
 let run_script ppf name args =
   override_sys_argv args;
-  Compmisc.init_path ~dir:(Filename.dirname name) ();
+  let filename =
+    match name with
+    | File name -> name
+    | Stdin | String _ -> "" in
```

Comment:
(This is a call to `Toploop.filename_of_input`.)

Reply:
i addressed the comments, locally, but after simple renaming the bytecode toplevel stopped working and tests started to fail.
however the version before renames shouldn't have worked as it was using constructors that didnt exits as `File` instead of `Toploop.File`. All in all i broke the thing, and i dont know why broken version works and fixed doesn't

Reply:
No worries. If you push the current state to your branch, I can have a look at the problem tomorrow.

Reply:
I had look at at it again today, and it looks like bytecode toplevel doesn't load anything before running, while native toplevel works as expected. And bytecode toplevel fails even when trying to declare a variable, as if it is unbound

Reply:
managed to reproduce minimize the issue: 
if function `filename_of_input` exitsts in toploop.mli the bytecode toplevel stops working, and cant bind a variable, as soon as i remove it from the interface it works again.

Reply:
okay, i needed to read comments. 
`Translmod.ml`
```ocaml
(* Compile a toplevel phrase *)

let toploop_ident = Ident.create_persistent "Toploop"
let toploop_getvalue_pos = 0 (* position of getvalue in module Toploop *)
let toploop_setvalue_pos = 1 (* position of setvalue in module Toploop *)
```